### PR TITLE
fix: nlb + autoscaling missing resource role

### LINKED ISSUE #3571: Getting error 'Stack ... does not exist' after updating to v1.18.1
Hello,
I started a copilot project with version 1.17.0 a few weeks ago. I have deployed 5 services without any problems. Today, I updated to version 1.18.1, created a new service with ```copilot svc init --name ...``` and wanted to deploy it. After copilot updated my staging environment to v1.9.0 I got the following error when I try ```copilot svc deploy --name ... --env staging```:

````
✘ Proposing infrastructure changes for stack <app_name>-staging-<svc_name>
✘ deploy service <svc_name> to environment staging: deploy service: check if changeset is empty: create change set copilot-3ea3048f-d3d7-4531-ae79-36a7b619e953 for stack <app_name>-staging-<svc_name>: ValidationError: Template error: instance of Fn::GetAtt references undefined resource CustomResourceRole
	status code: 400, request id: 6922ff6f-c789-4971-8f29-b6ca17346218: describe change set copilot-3ea3048f-d3d7-4531-ae79-36a7b619e953 for stack <app_name>-staging-<svc_name>: ValidationError: Stack [<app_name>-staging-<svc_name>] does not exist
	status code: 400, request id: 47937578-6294-4382-a3df-ad222ecbd4f4: describe stack: describe stack events for stack <app_name>-staging-<svc_name>: ValidationError: Stack [<app_name>-staging-<svc_name>] does not exist
	status code: 400, request id: abfdf3f6-2bc7-4ed8-84a1-1bd3e625e8aa

### PR DESCRIPTION
**Pull request by dannyrandall**
Fixes #3571 by refactoring out the `CustomResourceRole` and instead creates a role for each lambda individually. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/template/templates/workloads/partials/cf/alb.yml [hunk 1]
```diff
@@ -47,9 +47,34 @@ RulePriorityFunction:
     Handler: "index.nextAvailableRulePriorityHandler"
     Timeout: 600
     MemorySize: 512
-    Role: !GetAtt 'CustomResourceRole.Arn'
+    Role: !GetAtt "RulePriorityFunctionRole.Arn"
     Runtime: nodejs12.x
 
+RulePriorityFunctionRole:
```
Comment (iamhopaul123) [inline internal/pkg/template/templates/workloads/partials/cf/alb.yml:53]: we might need metadata for essential resources like [this](https://github.com/aws/copilot-cli/pull/3578/files#diff-4da05cf94ee9faf4c3183eeef2dd56e86646e1e43579df89d0fa2b3cdf394d9aL85-L86)

**Review by Lou1415926 — APPROVED** [review]
+1 to PH's!

File: internal/pkg/template/templates/workloads/partials/cf/alb.yml (hunk 1 above)
Comment (dannyrandall) [reply internal/pkg/template/templates/workloads/partials/cf/alb.yml:53]: added that back!

**Comment by paragbhingre** [comment]
LGTM 🎉

File: internal/pkg/template/templates/workloads/partials/cf/alb.yml [hunk 2]
```diff
@@ -47,9 +47,36 @@ RulePriorityFunction:
...
+  Metadata:
+    'aws:copilot:description': "An IAM Role for RulePriorityFunction"
```
Comment (efekarakus) [inline internal/pkg/template/templates/workloads/partials/cf/alb.yml]: nit:
```suggestion
    'aws:copilot:description': "An IAM Role to describe load balancer rules for assigning a priority"
```

File: internal/pkg/template/templates/workloads/partials/cf/autoscaling.yml [hunk 3]
```diff
@@ -23,9 +23,55 @@ DynamicDesiredCountFunction:
     Handler: "index.handler"
     Timeout: 600
     MemorySize: 512
-    Role: !GetAtt 'CustomResourceRole.Arn'
+    Role: !GetAtt 'DynamicDesiredCountFunctionRole.Arn'
     Runtime: nodejs12.x
 
+DynamicDesiredCountFunctionRole:
+  Metadata:
+    'aws:copilot:description': "An IAM Role for DynamicDesiredCountFunction"
```
Comment (efekarakus) [inline internal/pkg/template/templates/workloads/partials/cf/autoscaling.yml]: nit:
```suggestion
    'aws:copilot:description': "An IAM Role for describing number of running tasks in your ECS service"
```

**Review by efekarakus — APPROVED** [review]
so good!

**Pull request merged**